### PR TITLE
Add guile feedstock

### PR DESCRIPTION
**Pull request by ChristopherHogan**

### CODE REVIEW

**Comment by ChristopherHogan** [comment]
@conda-forge/core, this is green and ready for review.

File: recipes/guile/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./configure --prefix="${PREFIX}"
+make -j 2
+make install
```
Comment (ocefpaf) [inline recipes/guile/build.sh]: Is there a `make check` we can run here?

**Comment by scopatz** [comment]
@ChristopherHogan - please ping @conda-forge/staged-recipes :wink:

File: recipes/guile/build.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./configure --prefix="${PREFIX}"
+make -j 2
```
Comment (ocefpaf) [inline recipes/guile/build.sh]: Please use `$CPU_COUNT` instead of a hard-coded number of CPUs.

File: recipes/guile/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    - gettext
+
+test:
+  commands:
+    - guile -v
+
+about:
+  home: https://www.gnu.org/software/guile
+  license: LGPL-3.0
+  license_file: LICENSE
```
Comment (scopatz) [inline recipes/guile/meta.yaml]: Please set the `license_family: LGPL`

File: recipes/guile/build.sh (hunk 1 above)
Comment (ChristopherHogan) [reply recipes/guile/build.sh]: Mac fails on 1 test:
```
FAIL: threads.test: mutex-ownership: mutex with owner not retained (bug #27450)
```
The [referenced bug](http://savannah.gnu.org/bugs/?27450) was fixed in 2009.

Linux hangs while running srfi-18.test, which tests the thread library.  What's the best way to proceed?

**Comment by ChristopherHogan** [comment]
Depends on conda/conda-build#3373

File: recipes/guile/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://ftp.gnu.org/gnu/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 33b904c0bf4e48e156f3fb1d0e6b0392033bd610c6c9d9a0410c6e0ea96a3e5c
+  # patches:
+  #   - no_posix_test.patch  # [linux]
```
Comment (scopatz) [inline recipes/guile/meta.yaml]: Can you either remove this comment (and the patch) or explain in a comment under what conditions it should be uncommented? Thanks!

**Comment by ChristopherHogan** [comment]
This should be ready. I had to skip one test on linux, but the Red Hat rpm also skips it, as seen in the changelog [here](http://rpmfind.net/linux/RPM/opensuse/ports/tumbleweed/aarch64/guile-devel-2.2.4-2.1.aarch64.html) reproduced below:

<pre>
Changelog
* Tue Jan 15 2019 schwab@suse.de
  - remove broken prebuilt 32-bit big-endian objects
* Tue Nov 27 2018 jbrielmaier@suse.de
  - Update to version 2.2.4. [boo#1117121]
    Notable changes compared to 2.0.14:
    * speed
    * better thread and space-safety
    * improved locale support
    * dynamically expandable stacks
    * many small improvements and bug fixes
    This release brings incompatible changes to 2.0.x versions.
    More details: /usr/share/doc/packages/guile/NEWS
  - add patches:
    <b>* disable-posix-crypt-test.patch: test glibc EINVAL fails</b>
...
</pre>

**Pull request merged**